resources/build/run-required-test-builds.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+#
+# Determine if we need to do a build based on rules in
+# trigger-definitions.sh, rather than calculating changes in
+# TeamCity. If a build is needed, then we ask TeamCity to
+# start the build.
+#
+
+set -e
+set -u
+
+if [[ $# -lt 1 ]]; then
+  echo 'Usage: run-required-test-build.sh pull-request-number|branch'
+  echo '  where branch can be master, beta'
+  exit 1
+else
+  PRNUM="$1"
+fi
+
+#
+# This must be run under a TeamCity environment in order to receive all variables
+#
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null || readlink -f "${BASH_SOURCE[0]}")"
+. "$(dirname "$THIS_SCRIPT")/build-utils.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+. "$(dirname "$THIS_SCRIPT")/trigger-definitions.sh"
+. "$(dirname "$THIS_SCRIPT")/trigger-builds.sh"
+
+#
+# Definitions
+#
+
+available_platforms=(android ios linux mac web windows)
+
+#
+# Iterate through the platforms 'array' passed in and 
+# run builds associated with each platform found
+#
+
+function triggerTestBuilds() {
+  # Note: we always run builds for 'all' platforms
+  local platforms=( `echo "all $1"` )
+  local branch="$2"
+
+  for platform in "${platforms[@]}"; do
+    echo "# $platform: changes detected"
+    eval test_builds='(${'bc_test_$platform'[@]})'
+    for test_build in "${test_builds[@]}"; do
+      if [[ $test_build == "" ]]; then continue; fi
+      echo "  -- Triggering build configuration $test_build"
+      triggerBuild "$test_build" "$vcs_test" "$branch"
+    done
+  done
+}
+
+#
+# Following is not an error; this script can run against master/beta/stable
+# branches as well as pull requests
+#
+
+if [[ ! "$PRNUM" =~ ^[[:digit:]]+$ ]]; then
```

Comment:
If I understand correctly: `#case 1 - release builds`
- This section conditions against master/beta/stable changes and does not trigger for PR tests.